helm/cluster-openstack/templates/_helpers.tpl

```diff
@@ -70,6 +70,30 @@ room for such suffix.
   sudo: ALL=(ALL) NOPASSWD:ALL
 {{- end -}}
 
+{{- define "registryFiles" -}}
+{{- if .Values.registry.configure -}}
+- path: /etc/containerd/conf.d/registry-config.toml
+  permissions: "0600"
+  contentFrom:
+    secret:
+      name: {{ include "resource.default.name" $ }}-registry-configuration
+      key: registry-config.toml
+{{- end -}}
+{{- end -}}
+
+{{- define "registrySecretContent" -}}
+[plugins."io.containerd.grpc.v1.cri".registry]
+
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
+endpoint = ["https://registry-1.docker.io", "giantswarm.azurecr.io"]
+
+[plugins."io.containerd.grpc.v1.cri".registry.configs]
+[plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
+username = "{{ .Values.registry.username }}"
+password = "{{ .Values.registry.password }}"
```

Comment:
should we use token as we use for legacy clusters?
https://github.com/giantswarm/k8scloudconfig/pull/818/files#diff-dedaf91321442d2276905cb10322fa2b63252016c63df890e472c008fefe75f5R4